include/seqan3/core/algorithm/concept.hpp

```diff
@@ -76,6 +80,37 @@ struct config_id_accessor
     }
 
 public:
+    //!\brief Type alias for the internal enumeration type that represents the id.
+    template <typename config_t>
+    using id_type = std::remove_cvref_t<decltype(config_t::id)>;
+
+    /*!\brief Checks if two configuration types are compatible.
+     *
+     * \tparam config1_t The type of the first configuration to check against the second.
+     * \tparam config2_t The type of the second configuration.
+     *
+     * \details
+     *
+     * Uses the seqan3::detail::compatibility_table of the corresponding configuration element domain to check
+     * if both configurations can be combined.
+     */
+    template <typename config1_t, typename config2_t>
+    static constexpr auto is_compatible()
+    {
+        if constexpr (has_id_member<config1_t>(0) && has_id_member<config2_t>(0)) // needed for gcc-9 -stc=c++2a
+        {
+            if constexpr (std::same_as<id_type<config1_t>, id_type<config2_t>>) // needed for gcc-7 and gcc-8
+                return std::bool_constant<compatibility_table<id_type<config1_t>>[as_int<config1_t>]
+                                                                                 [as_int<config2_t>]>{};
+            else
+                return std::false_type{};
+        }
+        else
+        {
+            return std::false_type{};
+        }
```

Comment:
Could we use the

```c++
#if SEQAN3_WORKAROUND_GCC_... // gcc 9
  // gcc 9 code to make this work
#elif SEQAN3_WORKAROUND_GCC_... // gcc 7 and gcc 8
  // gcc 7 and 8 code to make this work
#else // ^^^ workaround / no workaround vvv
  // code that should work as it is
#endif

Reply:
Well, the whole thing should not be needed and work as it is. The concept checks already if both have  id_member and if they have the same type. Still for some reasons on some compilers I had to repeat the checks in the function as well. 
Are you sure, that cluttering it any further with workaround macros makes it more readable? I doubt that the compile time reduces much, since everything was instantiated at least once anyway.

Reply:
no, I purely think, what happens when we drop gcc 7, gcc 8 and so on. We should make the transition at that point as easy as possible by marking that we did a workaround, and we wanted to have a different solution.

Reply:
This is what the code would look like:
```cpp
#if     SEQAN3_WORKAROUND_GCC_9 // gcc 9
        if constexpr (has_id_member<config1_t>(0) && has_id_member<config2_t>(0)) // needed for gcc-9 -stc=c++2a
        {
#if     SEQAN3_WORKAROUND_GCC_less_than_9 // gcc 7 and gcc 8    
            if constexpr (std::same_as<id_type<config1_t>, id_type<config2_t>>) // needed for gcc-7 and gcc-8
#endif  SEQAN3_WORKAROUND_GCC_less_than_9 // ^^^ workaround / no workaround vvv
                return std::bool_constant<compatibility_table<id_type<config1_t>>[as_int<config1_t>]
                                                                                 [as_int<config2_t>]>{};
#if     SEQAN3_WORKAROUND_GCC_less_than_9 // gcc 7 and gcc 8    
            else
                return std::false_type{};
#endif SEQAN3_WORKAROUND_GCC_less_than_9 // ^^^ workaround / no workaround vvv
        }
        else
        {
            
            return std::false_type{};
        }
#endif  SEQAN3_WORKAROUND_GCC_9 // gcc 9
```
Not the fan of this, since the whole thing was supposed to not exist for any of the compilers.

Reply:
I mean the workaround is already below inside of the concept. I think that is expressive enough.

Reply:
How about:

```c++
#if SEQAN3_WORKAROUND_GCC_PIPEABLE_CONFIG_CONCEPT
    template <typename config1_t, typename config2_t>
    static constexpr auto is_compatible()
    {
        if constexpr (!(has_id_member<config1_t>(0) && has_id_member<config2_t>(0))) // needed for gcc-9 -stc=c++2a
            return std::false_type{};
#if (__GNUC__ < 9)
        else if constexpr (!std::same_as<id_type<config1_t>, id_type<config2_t>) // needed for gcc-7 and gcc-8
            return std::false_type{};
#endif
        else
            return std::bool_constant<compatibility_table<id_type<config1_t>>[as_int<config1_t>][as_int<config2_t>]>{};
    }
#endif // SEQAN3_WORKAROUND_GCC_PIPEABLE_CONFIG_CONCEPT
```
?

Reply:
let's see

Reply:
let's do your way... :)